test/solidity/Facets/WhitelistManagerFacet.t.sol

```diff
@@ -0,0 +1,774 @@
+        functionSelectors[0] = WhitelistManagerFacet.addToWhitelist.selector;
+        functionSelectors[1] = WhitelistManagerFacet
+            .removeFromWhitelist
+            .selector;
+        functionSelectors[2] = WhitelistManagerFacet
+            .batchAddToWhitelist
+            .selector;
+        functionSelectors[3] = WhitelistManagerFacet
+            .batchRemoveFromWhitelist
+            .selector;
+        functionSelectors[4] = WhitelistManagerFacet
+            .getWhitelistedAddresses
+            .selector;
+        functionSelectors[5] = WhitelistManagerFacet
+            .setFunctionApprovalBySelector
+            .selector;
+        functionSelectors[6] = WhitelistManagerFacet
+            .batchSetFunctionApprovalBySelector
+            .selector;
+        functionSelectors[7] = WhitelistManagerFacet
+            .isFunctionApproved
+            .selector;
+        functionSelectors[8] = WhitelistManagerFacet
+            .isAddressWhitelisted
+            .selector;
+        functionSelectors[9] = WhitelistManagerFacet
+            .getApprovedFunctionSelectors
+            .selector;
+
+        addFacet(diamond, address(whitelistMgr), functionSelectors);
+
+        // add AccessManagerFacet to be able to whitelist addresses for execution of protected functions
+        accessMgr = new AccessManagerFacet();
+
+        functionSelectors = new bytes4[](2);
+        functionSelectors[0] = accessMgr.setCanExecute.selector;
+        functionSelectors[1] = accessMgr.addressCanExecuteMethod.selector;
+        addFacet(diamond, address(accessMgr), functionSelectors);
+
+        accessMgr = AccessManagerFacet(address(diamond));
+        whitelistMgr = WhitelistManagerFacet(address(diamond));
+    }
+
+    function test_SucceedsIfOwnerAddsAddress() public {
+        vm.startPrank(USER_DIAMOND_OWNER);
+
+        whitelistMgr.addToWhitelist(address(c1));
+        address[] memory approved = whitelistMgr.getWhitelistedAddresses();
+        assertEq(approved[0], address(c1));
+
+        vm.stopPrank();
+    }
+
+    function test_SucceedsIfOwnerRemovesAddress() public {
+        vm.startPrank(USER_DIAMOND_OWNER);
+
+        vm.expectEmit(true, true, true, true);
+        emit AddressWhitelisted(address(c1));
+
+        whitelistMgr.addToWhitelist(address(c1));
+
+        vm.expectEmit(true, true, true, true);
+        emit AddressRemoved(address(c1));
+
+        whitelistMgr.removeFromWhitelist(address(c1));
+
+        vm.stopPrank();
+
+        address[] memory approved = whitelistMgr.getWhitelistedAddresses();
+        assertEq(approved.length, 0);
+    }
+
+    function test_SucceedsIfOwnerBatchAddsAddresses() public {
+        vm.startPrank(USER_DIAMOND_OWNER);
+
+        address[] memory addresses = new address[](3);
+        addresses[0] = address(c1);
+        addresses[1] = address(c2);
+        addresses[2] = address(c3);
+        whitelistMgr.batchAddToWhitelist(addresses);
```

Comment:
same here, could move this batchAdd part into its own test case and call it from others that need it for setup

Reply:
ok, created private helper function for it